### ISSUE TITLE
KotlinJvmAndroidCompilation with name 'debug' not found.

### ISSUE THREAD

**Issue by RinOrz** [diffplug/spotless#1022]
- Gradle Version: 7.3.1
- Kotlin Version: 1.6.0
- Android Gradle Plugin Version: 7.0.3

> When I upgraded from **6.0.2** to **6.0.3/6.0.4**, an Android error appeared: 

**Stacktrace (https://github.com/meowool-toolkit/gradle-toolkit/runs/4447098883?check_suite_focus=true#step:4:80)**
```
* What went wrong:
A problem occurred configuring project ':integration-testing:multiplatform'.
> Failed to notify project evaluation listener.
   > java.lang.IllegalArgumentException: Cannot change attributes of dependency configuration ':integration-testing:multiplatform:androidDebugCompileOnly' after it has been resolved
   > KotlinJvmAndroidCompilation with name 'debug' not found.
   > KotlinJvmAndroidCompilation with name 'debug' not found.
   > KotlinJvmAndroidCompilation with name 'debug' not found.

Cause 1: com.android.build.gradle.internal.crash.ExternalApiUsageException: java.lang.IllegalArgumentException: Cannot change attributes of dependency configuration ':integration-testing:multiplatform:androidDebugCompileOnly' after it has been resolved
	at com.android.build.gradle.internal.ApiObjectFactory.create(ApiObjectFactory.java:124)
	at com.android.build.gradle.internal.plugins.BasePlugin.createAndroidTasks(BasePlugin.java:791)
	at com.android.build.gradle.internal.profile.NoOpAnalyticsConfiguratorService.recordBlock(NoOpAnalyticsConfiguratorService.kt:48)
	at com.android.build.gradle.internal.plugins.BasePlugin.lambda$createTasks$13(BasePlugin.java:650)
	at com.android.build.gradle.internal.crash.CrashReporting$afterEvaluate$1.execute(crash_reporting.kt:37)

Caused by: java.lang.IllegalArgumentException: Cannot change attributes of dependency configuration ':integration-testing:multiplatform:androidDebugCompileOnly' after it has been resolved
	at org.gradle.api.internal.attributes.ImmutableAttributeContainerWithErrorMessage.attribute(ImmutableAttributeContainerWithErrorMessage.java:57)
	at org.jetbrains.kotlin.gradle.plugin.KotlinTargetConfiguratorKt.usesPlatformOf(KotlinTargetConfigurator.kt:513)
	at org.jetbrains.kotlin.gradle.plugin.ProjectLocalConfigurationsKt.setupAsLocalTargetSpecificConfigurationIfSupported(ProjectLocalConfigurations.kt:49)
	at org.jetbrains.kotlin.gradle.plugin.AbstractKotlinTargetConfigurator$Companion.defineConfigurationsForCompilation(KotlinTargetConfigurator.kt:326)
	at org.jetbrains.kotlin.gradle.plugin.Android25ProjectHandler.setUpDependencyResolution(Android25ProjectHandler.kt:115)
	at org.jetbrains.kotlin.gradle.plugin.AbstractAndroidProjectHandler$configureTarget$2.invoke(KotlinPlugin.kt:824)
	at org.jetbrains.kotlin.gradle.plugin.AbstractAndroidProjectHandler$configureTarget$2.invoke(KotlinPlugin.kt:815)

Cause 2: org.gradle.api.UnknownDomainObjectException: KotlinJvmAndroidCompilation with name 'debug' not found.
	at org.gradle.api.internal.DefaultNamedDomainObjectCollection.createNotFoundException(DefaultNamedDomainObjectCollection.java:504)
	at org.gradle.api.internal.DefaultNamedDomainObjectCollection.getByName(DefaultNamedDomainObjectCollection.java:333)
	at org.jetbrains.kotlin.gradle.plugin.AbstractAndroidProjectHandler$configureTarget$3$1.invoke(KotlinPlugin.kt:836)
	at org.jetbrains.kotlin.gradle.plugin.AbstractAndroidProjectHandler$configureTarget$3$1.invoke(KotlinPlugin.kt:835)
```

May be related to the update here:
https://github.com/diffplug/spotless/blob/main/plugin-gradle/CHANGES.md#603---2021-12-06

Repository:
https://github.com/meowool-toolkit/gradle-toolkit/tree/spotless_1022

**Comment by nedtwigg** [comment]
The stacktrace doesn't appear to be related so Spotless.

I cloned your repo, checked-out `main` (which is using Spotless 6.0.4), and ran `./gradlew spotlessApply`. It completed normally, except that a few files had formatting changes.

Then I checked-out `spotless_1022` and got the error you describe above. But the only change between `main` and `spotless_1022` has nothing to do with Spotless.

https://github.com/meowool-toolkit/gradle-toolkit/commit/09b8647e5c6a87b1a5b44f7aae6824e9f2aa8752

Feel free to add more info, but this looks unrelated to Spotless.
